tests/bolt_str_copy_bench.c

```
@@ -0,0 +1,72 @@
+int main() {
+    size_t lengths[] = {16, 128, 1024, 1024 * 1024};
+    const char* labels[] = {"16B", "128B", "1KB", "1MB"};
+    int num_lengths = 4;
+
+    printf("--- String Copy Benchmark ---\n");
+
+    for (int i = 0; i < num_lengths; i++) {
+        size_t len = lengths[i];
+        char* src = (char*)malloc(len + 1);
+        char* dest = (char*)malloc(len + 1);
+        memset(src, 'A', len);
```

Cppcheck (2.19.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)